src/reader/mod.rs

```diff
@@ -21,42 +20,23 @@ mod parser;
 mod config;
 mod events;
 
+mod error_impl;
+
+#[derive(Debug)]
+pub enum ErrorKind {
```

Comment:
You don't need to name modules, they can be private. Reexporting names is a very common practice in Rust.

```rust
mod reader {

    mod error {
        pub struct Error { ... }
        pub enum ErrorKind { ... }
    }

    pub use self::error::{Error, ErrorKind};
}
```
Now from the outside it would look as if `Error` and `ErrorKind` are defined in `reader` itself, while in fact they are defined in `reader::error`, which won't even be visible because it is not `pub`.